helix-term/src/application.rs

```diff
@@ -974,17 +978,66 @@ impl Application {
                             &params.edit,
                         );
 
-                        Ok(json!(lsp::ApplyWorkspaceEditResponse {
+                        Some(Ok(json!(lsp::ApplyWorkspaceEditResponse {
                             applied: true,
                             failure_reason: None,
                             failed_change: None,
-                        }))
+                        })))
                     }
                     Ok(MethodCall::WorkspaceFolders) => {
                         let language_server =
                             self.editor.language_servers.get_by_id(server_id).unwrap();
 
-                        Ok(json!(language_server.workspace_folders()))
+                        Some(Ok(json!(language_server.workspace_folders())))
+                    }
+                    Ok(MethodCall::ShowMessageRequest(params)) => {
+                        if let Some(actions) = params.actions {
+                            let call_id = id.clone();
+                            let icon = match params.typ {
+                                MessageType::ERROR => "❗",
+                                MessageType::WARNING => "⚠️",
+                                MessageType::INFO => "ℹ️",
+                                _ => "📝",
+                            };
+                            let message = format!("{}: {}", icon, &params.message);
```

Comment:
Emoji icons look out of place in the UI - we use unicode icons sparingly but not emoji AFAIK.

Usually we color spans with hint/info/warning/error from the theme (for exmaple the diagnostics pickers) but that isn't possible with the file preview here. I would prefer nvim's style without an indication of the severity, or a text marker like `WARNING: <message>`/`ERROR: <message>`

Reply:
I went with the text marker